core/src/main/java/io/grpc/internal/DnsNameResolver.java

```diff
@@ -253,6 +286,27 @@ public void run() {
       }
     };
 
+  /** Returns value of security property of network address cache ttl. */
+  private long getNetworkAddressCacheTtl() {
+    String cacheTtlPropertyValue = Security.getProperty(NETWORKADDRESS_CACHE_TTL_SECURITY_PROPERTY);
+    long cacheTtl = DEFAULT_NETWORK_CACHE_TTL;
```

Comment:
Optional: but I strong recommend converting everything to nanoseconds on API boundaries.  It makes the code base consistent, and avoids converting in the internal code paths.